controller.go

```diff
@@ -27,6 +27,7 @@ const (
 	defaultHTTPRedirectToHTTPS    = "false"
 	defaultCertTTL                = "1h"
 	customTagFilterEnvVarName     = "CUSTOM_FILTERS"
+	defaultInternalDomains        = "*.ingress.cluster.local"
```

Comment:
Does this match subdomains, too?
I bet it's a string suffix search, so we could also re-use `"*" + kubernetes.DefaultClusterLocalDomain`, which is `.cluster.local`

Reply:
Done, I set it to default to `"*" + kubernetes.DefaultClusterLocalDomain` but didn't reutilize the `--cluster-local-domain` flag as I believe `--internal-domains` can be more powerful.

It accepts any config from the [HostHeaderConfig], multiple values and I think we can in the future create some config that does not deny such domains on [internal load balancers][internal-lb]. It might not be useful (yet) in Zalando's setup, but can be useful in other scenarios. As we will rely on the defaults, it won't make the config more difficult or verbose.

[HostHeaderConfig]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-listenerrule-hostheaderconfig.html
[internal-lb]: https://github.com/zalando-incubator/kube-ingress-aws-controller#create-an-internal-load-balancer

Reply:
I agree!